save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/SandboxView.kt

```diff
@@ -65,16 +65,31 @@ external interface SandboxViewState : State, HasSelectedMenu<ContestMenuBar> {
      */
     var codeText: String
 
+    /**
+     * Code from backend
+     */
+    var savedCodeText: String
```

Comment:
WDYT about extracting an editor window into a separate component that would store a draft (==unsaved) version of its text in its own state and update parent component's state on save? This way you could avoid duplicating all three fields in this component's state as well as always making 3 requests on save.

Reply:
We have two components on the same level (children of the entire editor component): editor itself and it's toolbar. I am not sure how to pass event from one component to another horizontally.
But Idea is very nice as far as more logic that could be encapsulated is actually being encapsulated

Reply:
Oh I got it

Reply:
Done

Reply:
My original suggestion was to remove text pairs (regular / saved version) so that unsaved versions are encapsulated in a new component and regular versions in the parent component are updated on save

Reply:
I am not sure if your variant would be reusable enough as far as we wanted to make diktat-demo be a part of saveourtool

Reply:
I am not sure if your variant would be reusable enough as far as we wanted to make diktat-demo be a part of saveourtool

Reply:
I think it's a rather far perspective actually

Reply:
Then the question of sending data rases. We might want to either send one of drafts or all of them. Also it is not that clear how to pass isModified flag from codeEditorComponent to toolbar. Seems like we might need even more states in that case.

Reply:
Now you have: 3 tabs, each has 2 text fields in state. What if we add more state variables for each tab? You'll need to add 3 more overall. What if we add more tabs?

What I suggest: 1 text field in state of parent, 1 field for isModified for each tab. Each instance of CodeEditorTab (or whatever you call it) receives text as props and a callback to update parent's isModified. Draft text and whatever state we may add in the future lives only inside CodeEditorTab. Toolbar can receive callbacks to access/modify parent's isModified too. I think this is more scalable and less coupled with the specific tabs we may have.

Reply:
Ok, got what you want. In this case I would like to add **another** one component, but probably in next PR as this one is getting bigger and bigger